enhancements/ci-enhancement.md

```diff
@@ -0,0 +1,73 @@
+# Summary
+
+This enhancement proposal is to improve the overall performance and functionality of the operator-sdk CI and testing. There are a number of issues currently plaguing our system, some of which can be handled by switching to a higher tier of compute service and some of which will require serious revisions to our testing process.
+
+# Motivations
+
+The primary motivation for these enhancements is to improve developer productivity. Much time is lost in slow CI runs, inconsistent flakes and failures in CI, and a general difficulty in debugging said CI runs. 
+
+The primary issues are the following:
+
+Extremely slow TravisCI runs
+
+Can take upwards of 30mins for a single set of e2e tests to execute
+
+Unit test coverage is subpar
+
+Increased unit testing would also allow us to alleviate some of the e2e testing load
+
+Testing is stitched together across a number of bash scripts
+
+Very difficult to debug specific points of failure
+
+No unified structure for e2e testing
+
+General flakes and failures
+
+Travis passes but GitHub does not allow merging
```

Comment:
this happened a few times but i think it is important to emphasize that this problem is intermittent and is not frequent.